operator/controllers/metrics/keptnmetric_controller.go

```diff
@@ -61,12 +73,61 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	return ctrl.Result{}, nil
+	if time.Now().Before(metric.Status.LastUpdated.Add(time.Second * time.Duration(metric.Spec.FetchIntervalSeconds))) {
```

Comment:
it's a long expression 😅 would it make sense to split it?
```suggestion
	if deadline := metric.Status.LastUpdated.Add(time.Second * time.Duration(metric.Spec.FetchIntervalSeconds)); time.Now().Before(deadline) {
```